pkg/controller/populators/util.go

```diff
@@ -92,6 +97,41 @@ func getPopulatorIndexKey(apiGroup, kind, namespace, name string) string {
 	return fmt.Sprintf("%s/%s/%s/%s", apiGroup, kind, namespace, name)
 }
 
+func claimReadyForPopulation(ctx context.Context, c client.Client, pvc *corev1.PersistentVolumeClaim) (bool, string, error) {
+	if pvc.Spec.StorageClassName == nil {
+		// maybe storageclass will get assigned later
+		return false, "", nil
+	}
+
+	if *pvc.Spec.StorageClassName == "" {
+		return false, "", fmt.Errorf("PVC %s/%s has empty storage class name", pvc.Namespace, pvc.Name)
```

Comment:
so incase storageclass name is empty we want to return error and not go by the default storageclass?

Reply:
default storageclass is irrelevent.  PVC is created and storageclass has been assigned.  This is not like pvc definition in a datavolume where there is flexibility